lib/cn/kvs_mblk_desc.c

```diff
@@ -91,8 +91,9 @@ mblk_madvise_pages(const struct kvs_mblk_desc *md, size_t pg, size_t pg_cnt, int
         return 0;
 
     ra_pages = (advice == MADV_WILLNEED) ? md->ra_pages : pg_cnt;
+    ev(!ra_pages);
 
-    for (size_t pg_end = pg + pg_cnt; pg < pg_end; pg += chunk) {
+    for (size_t pg_end = pg + pg_cnt; ra_pages && (pg < pg_end); pg += chunk) {
```

Comment:
You already have the `ev` for `ra_pages==0`, so why not return there?
```
if (ev(!ra_pages))
    return 0;
```

Reply:
Could do.. I thought about that when coding this but decided a save one line :-) Will change...